pkg/pubsub/pubsub_client.go

```diff
@@ -0,0 +1,119 @@
+package pubsub
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/dapr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func init() {
+	flag.Var(pubSubs, "pub-sub-tool", "Preferred pub-sub tool, e.g dapr, rabbitmq. This flag can be declared more than once. Omitting will default to not using any pub sub tool.")
+}
+
+var pubSubs = newPubSubSet(map[string]PubSub{
+	dapr.Name: &dapr.Dapr{},
+},
+)
+
+// PubSub is the interface that wraps pubsub methods.
+type PubSub interface {
+	// Publish single message over a specific topic/channel
+	Publish(data interface{}, topic string) error
+
+	// Publish a batch of messages over a specific topic/channel
+	PublishBatch(data interface{}, topic string) error
+
+	// Initiate the pub sub client
+	NewClient(ctx context.Context, client client.Client) error
```

Comment:
Why does this take client.Client? Per above, assuming the K8s client is available would make using this in `gator test` difficult.

Also, NewClient() should probably be a method that returns an object that satisfies the PubSub interface -- usually interfaces to not also describe their own constructors.

Lastly, New()-type methods in golang usually return a newly constructed object.

Reply:
Modified the code to define proper constructor